src/main/java/org/openrewrite/java/logging/jboss/LoggerLevelArgumentToMethod.java

```diff
@@ -0,0 +1,124 @@
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.logging.jboss;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class LoggerLevelArgumentToMethod extends Recipe {
+    private static final MethodMatcher LOG_MATCHER = new MethodMatcher("org.jboss.logging.Logger log(..)", true);
+    private static final MethodMatcher LOGF_MATCHER = new MethodMatcher("org.jboss.logging.Logger logf(..)", true);
+    private static final MethodMatcher LOGV_MATCHER = new MethodMatcher("org.jboss.logging.Logger logv(..)", true);
+
+    @Override
+    public String getDisplayName() {
+        return "Replace JBoss Logging Level arguments with the corresponding eponymous level method calls";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace calls to `Logger.log(Level, ...)` with the corresponding eponymous level method calls. For example `Logger.log(Level.INFO, ...)` to `Logger.info(...)`.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        JavaVisitor<ExecutionContext> javaVisitor = new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation mi, ExecutionContext ctx) {
+                J.MethodInvocation m = super.visitMethodInvocation(mi, ctx);
+                if (!(LOG_MATCHER.matches(m) || LOGF_MATCHER.matches(m) || LOGV_MATCHER.matches(m))) {
+                    return m;
+                }
+                List<Expression> args = m.getArguments();
+                Expression firstArgument = requireNonNull(args.get(0));
+                Expression secondArgument = requireNonNull(args.get(1));
+
+                String formatted = "";
+                if (LOGF_MATCHER.matches(m) || LOGV_MATCHER.matches(m)) {
+                    if (TypeUtils.isAssignableTo("java.lang.String", firstArgument.getType())) {
+                        // `logger.logf(String fqcn ...)` and `logger.logv(String fqcn ...)` don't have a logger.level() equivalent.
```

Comment:
I would use the [Method patterns](https://docs.openrewrite.org/reference/method-patterns) syntax in the explanatory comments. So in this case:


```suggestion
                        // `logf(String, ..)` and `logv(String, ..)` don't have a logger.level() equivalent
```

I don't know for sure, but I think people will understand what you mean by these comments quicker!